crates/wasmtime/src/runtime.rs

```diff
@@ -694,13 +696,39 @@ impl Engine {
         debug_builtins::ensure_exported();
         Engine {
             config: Arc::new(config.clone()),
+            compiler: Arc::new(Engine::build_compiler(config)),
         }
     }
 
     /// Returns the configuration settings that this engine is using.
     pub fn config(&self) -> &Config {
         &self.config
     }
+
+    pub(crate) fn compiler(&self) -> &Compiler {
+        &self.compiler
+    }
+
+    fn build_compiler(config: &Config) -> Compiler {
+        let isa = native::builder().finish(settings::Flags::new(config.flags.clone()));
+        Compiler::new(
+            isa,
+            config.strategy,
+            config.cache_config.clone(),
+            config.tunables.clone(),
+        )
+    }
+
+    /// Returns whether the engine `a` and `b` refer to the same configuration.
+    pub fn same(a: &Engine, b: &Engine) -> bool {
+        Arc::ptr_eq(&a.config, &b.config)
+    }
+}
+
+impl Default for Engine {
+    fn default() -> Engine {
+        Engine::new(&Config::default())
```

Comment:
Could `Default for Store` delegate to this now that it exists?

Reply:
Not sure how to do it: `Engine::new` and `Store::new` have some additional logic that needs to be run.